src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -1688,6 +1688,30 @@ namespace InternalHeatGains {
                         ShowContinueError(state, "No matching Zone Return Air Node found.");
                         ErrorsFound = true;
                     }
+                    // Set exhaust air node number
+                    state.dataHeatBal->Lights(Loop).ZoneExhaustNodeNum = 0;
+                    if (!state.dataIPShortCut->lAlphaFieldBlanks(8)) {
+                        if (state.dataHeatBal->LightsObjects(Item).ZoneListActive) {
+                            ShowSevereError(state,
+                                            RoutineName + CurrentModuleObject + "=\"" + state.dataHeatBal->Lights(Loop).Name +
+                                                "\": " + state.dataIPShortCut->cAlphaFieldNames(8) + " must be blank when using a ZoneList.");
+                            ErrorsFound = true;
+                        } else {
+                            state.dataHeatBal->Lights(Loop).ZoneExhaustNodeNum =
+                                UtilityRoutines::FindItemInList(AlphaName(8), state.dataLoopNodes->NodeID, state.dataLoopNodes->NumOfNodes);
+                            if (state.dataHeatBal->Lights(Loop).ZoneExhaustNodeNum == 0) {
```

Comment:
This is not checking to see if the node is a zone exhaust node. It's just checking if `AlphaName(8)` is in the list of nodes.
This really should be more like:
```
bool exhaustNodeError = false;
state.dataHeatBal->Lights(Loop).ZoneExhaustNodeNum = GetOnlySingleNode( state,
                                                                                             AlphaName(8),
                                                                                             exhaustNodeError ,
                                                                                             CurrentModuleObject,
                                                                                             LightsObjects(Item).Name),
                                                                                             DataLoopNode::NodeFluidType::Air,
                                                                                             DataLoopNode::NodeConnectionType::ZoneExhaust,
                                                                                             NodeInputManager::compFluidStream::Primary,
                                                                                             ObjectIsNotParent);
if (!exhaustNodeError){ //GetOnlySingleNode will throw error messages if the is a NodeList Name and for other issues
   exhaustNodeError = DataZoneEquipment::VerifyLightsExhaustNodeForZone(
                            state, state.dataHeatBal->Zone(state.dataHeatBal->Lights(Loop).ZonePtr).Name, state.dataHeatBal->Lights(Loop).ZoneExhaustNodeNum);
}
if (exhaustNodeError){
 . . . . error message about invalid exhaust node
}
```
`DataZoneEquipment::VerifyLightsExhaustNodeForZone` would be a new `bool` function modeled after `DataZoneEquipment::GetReturnNumForZone`. But it would just look for a matching zone exhaust node num and return false if it's not found.

Reply:
@mjwitte I think "state.dataHeatBal->Lights(Loop).ZoneExhaustNodeNum = GetOnlySingleNode( state,..." is necessary. However, a new bool function may not be necessary, because ZoneNum is given before, and a single line check should be enough as:

                            state.dataHeatBal->Lights(Loop).ZoneExhaustNodeNum =
                                UtilityRoutines::FindItemInList(AlphaName(8), state.dataLoopNodes->NodeID, state.dataLoopNodes->NumOfNodes);

what do you think?

Reply:
All that line does is see if `AlphaName(8)` exists in the list of node names. It doesn't check if that node name is actually a zone exhaust node.